metrotranscom/doorway
# feat: require terms and conditions acceptance at log-in

This PR addresses [#(803)](https://app.zenhub.com/workspaces/bloom-5dc32d7144bd400001315dac/issues/gh/metrotranscom/doorway/803)


## Description

For SMC users who are being ported over, they will need to agree to the BAHFA terms of service before logging into their accounts. When they attempt to login, either through pwdless or with email and password, a modal should appear prompting them to agree. If they agree they will be signed in and redirected to the dashboard page. All other flows should not be impacted.

## How Can This Be Tested/Reviewed?
For a public account, manually mark agreedToTermsOfService to FALSE in the db.

Attempt to sign in to the public site with username and password. Modal should appear. Accepting the terms will redirect you to the account dashboard.

Attempt to sign in to the public site with pwdless. Input code emailed to you. Modal should appear. Accepting the terms will redirect you to the account dashboard.

Check other flows are not impacted:
- Create account
- Log-in with agreedToTermsOfService as TRUE
- Forgot password
- Confirm account (partners)

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
